### PR TITLE
DOC: Fix the interpolation formulae for quantile and percentile

### PR DESCRIPTION
**Pull request by deego**
…tile() and percentile().

Fix the interpolation formulae in the docs which led to absurd results.  For quantile() and percentile().

Example, for median of a = [1,2,10,11], you expect to obtain i+g=2.5 for method = linear (or weibull, or hazen, or median_unbiased or normal_unbiased).

Instead, you obtain a /negative/ index.

The correted formula is:

i + g = q * (n - alpha - beta + 1 ) + alpha

Notice among other things that n belongs in the numerator, not the denominator! 

As a check, the corrected formula does lead to the correct index 2.5 for each of the cases above.

MYSTERY: Surely the original formula was the result of a small typo/thinko?  Then, why does the correction look so completely different?

RESOLUTION OF MYSTERY: 
Take our formula, massage it, and swap q with (i+g), and you end up with the original formula.
In other words, the original author of the doc. simply confused their percentile with their index halfway through the creation of the doc. Then, they massaged it to isolate (i+g) on the left.

### CODE REVIEW

**Comment by deego** [comment]
Sorry, the comment was messed up. (I have never used git before.)

The (doc) formulae for quantile() and percentile() were messed up. 

I fixed these.

**Comment by deego** [comment]
Seberg, 
Yeah, great point about 1-based indexing. 

As far as the formula itself is concerned, I'd suggest leaving it alone:
(a) Math people are used to 1-based indexing, and I think this 1-based formula might correspond to the one in the various referenced papers. 
(b) This notion of (i+g) carries throughout the module , if I remember correctly. So, if we change it at one place, we should change it everywhere.  

But, yeah, if you see fit, wouldn't hurt to add an additional note reminding us that the actual python code subtracts an additional 1 from the index since python is 0-based. 

Such a note, if added, might probably fit better in the other section that introduces the concept of fractional index.

**Comment by seberg** [comment]
Ok, let me just put it in. Thanks!  (I still would like a small tweak to mention 1-based in the text somewhere, but not immediately sure where and this is clearly better than a simply incorrect formula)

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @deego! We hope you stick around! Your choices aren’t limited to programming – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion
